components/automate-cli/cmd/chef-automate/certRotate.go

```diff
@@ -0,0 +1,275 @@
+}
+
+func certRotate(cmd *cobra.Command, args []string) error {
+	rootCaPath := certFlags.rootCa
+	privateCertPath := certFlags.privateCert
+	publicCertPath := certFlags.publicCert
+	fileName := "cert.toml"
+
+	if privateCertPath == "" || publicCertPath == "" {
+		return errors.New("Please provide public and private cert paths")
+	}
+
+	var rootCA []byte
+	var err error
+	if sshFlag.postgres {
+		if rootCaPath == "" {
+			return errors.New("Please provide rootCA path")
+		}
+		rootCA, err = ioutil.ReadFile(rootCaPath) // nosemgrep
+		if err != nil {
+			return status.Wrap(
+				err,
+				status.FileAccessError,
+				fmt.Sprintf("failed reading data from file: %s", err),
+			)
+		}
+	}
+
+	privateCert, err := ioutil.ReadFile(privateCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err),
+		)
+	}
+
+	publicCert, err := ioutil.ReadFile(publicCertPath) // nosemgrep
+	if err != nil {
+		return status.Wrap(
+			err,
+			status.FileAccessError,
+			fmt.Sprintf("failed reading data from file: %s", err),
+		)
+	}
+
+	f, err := os.Create(fileName)
+	if err != nil {
+		log.Fatal(err)
+	}
+	infra, err := getAutomateHAInfraDetails()
+	if err != nil {
+		return err
+	}
+	sshUser := infra.Outputs.SSHUser.Value
+	sskKeyFile := infra.Outputs.SSHKeyFile.Value
+	sshPort := infra.Outputs.SSHPort.Value
+
+	if isA2HARBFileExist() {
+		if sshFlag.automate {
+			config := fmt.Sprintf(`
+			[[load_balancer.v1.sys.frontend_tls]]
+				cert = """%v"""
+				key = """%v"""
+			[[global.v1.frontend_tls]]
+				cert = """%v"""
+				key = """%v"""
+		`, string(publicCert), string(privateCert), string(publicCert), string(privateCert))
+
+			_, err = f.Write([]byte(config))
+			if err != nil {
+				log.Fatal(err)
+			}
+			f.Close()
+			automateIps := infra.Outputs.AutomatePrivateIps.Value
+			scriptCommands := fmt.Sprintf(FRONTEND_COMMANDS, fileName, dateFormat)
+			for i := 0; i < len(automateIps); i++ {
+				connectAndExecuteCommandOnRemote(sshUser, sshPort, sskKeyFile, automateIps[i], fileName, scriptCommands)
+			}
+
+		} else if sshFlag.postgres {
+			config := fmt.Sprintf(`
+			 [ssl]
+			 enable = true
+			 ssl_key = """%v"""
+			 ssl_cert = """%v"""
+			 issuer_cert = """%v"""
+			`, string(privateCert), string(publicCert), string(rootCA))
+
+			_, err = f.Write([]byte(config))
+			if err != nil {
+				log.Fatal(err)
+			}
+			f.Close()
+			const remoteService string = "postgresql"
+			scriptCommands := fmt.Sprintf(BACKEND_COMMAND, remoteService, dateFormat, fileName)
+			connectAndExecuteCommandOnRemote(sshUser, sshPort, sskKeyFile, infra.Outputs.PostgresqlPrivateIps.Value[0], fileName, scriptCommands)
+		}
+	}
+	return nil
+}
+
+func connectAndExecuteCommandOnRemote(sshUser string, sshPort string, sshKeyFile string, hostIP string, tomlFilePath string, remoteCommands string) {
```

Comment:
Rebase current branch and reuse the function `ConnectAndExecuteCommandOnRemote` in config.go for executing script in remote